src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs

```diff
@@ -277,23 +277,20 @@ public ObservableCollection<string> PackagePathsForInstall
         {
             get
             {
+                var allowedFileExtensions = new string[] { ".dll", ".ds" };
                 if (packagePathsForInstall == null || !packagePathsForInstall.Any())
                 {
                     var programDataPath = Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData);
                     // Filter Builtin Packages and ProgramData paths from list of paths for download
                     var customPaths = preferenceSettings.CustomPackageFolders.Where(
                         x => x != DynamoModel.BuiltInPackagesToken && !x.StartsWith(programDataPath));
+                    //filter out paths that have extensions ending in .dll or .ds
+                    var directoryPaths = customPaths.Where(path => !(Path.HasExtension(path) && allowedFileExtensions.Contains(Path.GetExtension(path).ToLower())));
 
                     packagePathsForInstall = new ObservableCollection<string>();
-                    foreach (var path in customPaths)
+                    foreach (var path in directoryPaths)
                     {
-                        var attr = File.GetAttributes(path);
-
-                        // Add only directory paths
-                        if ((attr & FileAttributes.Directory) == FileAttributes.Directory)
```

Comment:
Did this check not filter out the file paths? Shouldn't we be filtering out all file paths, not just .dll and .ds?

Reply:
> @mjkkirschner is there any benefit to adding the dll file path to the list of package search paths or is it simply to align it with the old dialog? How about adding only the directory _containing_ the dll to this list of search paths?

We need to add the .dll path directly to the search paths - it's for loading that binary directly, like we loaded it using `ImportLibrary`


> Did this check not filter out the file paths? Shouldn't we be filtering out all file paths, not just .dll and .ds, as this list is supposed to only contain directories?

This check relies on the path actually existing, the AC we decided on yesterday during planning was not to remove paths which no longer exist from the dropdown - instead we'll just recreate them. So we can't check if a path is a directory using this lookup because the path may not exist. As I explained yesterday - `Path.HasPath` will return true for a path like `.... Dynamo Core/ 2.12` - we don't want to filter that. Because we limit the paths that can be selected to be directories, .dlls, or .ds files this should be safe. It *won't* be safe if a user manually edits the custom package paths to include some other file type.